src/common/JsonFileManager.ts

```diff
@@ -31,6 +32,16 @@ export default class JsonFileManager<T> {
         });
     }
 
+    writeToFile(): Promise<void> {
+        const json = JSON.stringify(this.json, undefined, 2);
+        if (this.saving) {
+            this.saving = this.saving.then(() => this.write(json));
+        } else {
+            this.saving = this.write(json);
+        }
+        return this.saving;
```

Comment:
So we need this return?

Reply:
Yes, we still want the method to return a promise in case it wants to `await` it.